### ISSUE TITLE
Scala 3 compilation fails when using sbt 1.5.5+ 

### ISSUE THREAD

**Issue by Marcus-Rosti** [akka/akka-grpc#1471]
The best test case is to just change https://github.com/raboof/akka-grpc/blob/42062a151c0458817751a3ce1edeea44f99c37db/sbt-plugin/src/sbt-test/scala3/01-basic-client-server/project/build.properties#L1 to 1.5.5

Not sure why, but noticed this when trying to migrate a project. I think sbt is trying to pull in a conflicting version of scalapb

**Comment by raboof** [comment]
Interesting, thanks for the ping! Indeed I see both _3 and _2.13 dependencies:

```
Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/tmp/sbt_7164d611/"), "sbt_7164d611"):
[info] [error]    org.scala-lang.modules:scala-collection-compat _3, _2.13
[info] [error]    com.thesamet.scalapb:lenses _3, _2.13
[info] [error]    com.thesamet.scalapb:scalapb-runtime _3, _2.13
```

I guess this is really a conflict in theory, as akka-grpc-runtime depends on scalapb and is `_2.13`, but the generated code depends on scalapb as well and is `_3`. I could imagine there is no actual conflict "in practice", though, which is why it worked with sbt 1.5.4.

Unfortunately it seems adding an explicit:

```
dependencyOverrides ++= Seq(
  ("com.thesamet.scalapb" %% "lenses" % "0.11.5")
    .cross(CrossVersion.for3Use2_13)
)
```

doesn't seem to help, so we'd have to figure out where the `_3` dependency still comes from.

**Comment by raboof** [comment]
> Yes, this was fixed in sbt 1.5.5 [sbt/sbt#6578](https://github.com/sbt/sbt/issues/6578)
> 
> as a workaround I currently use
> 
> ```
> excludeDependencies ++= Seq(
>   "com.thesamet.scalapb"   % "scalapb-runtime_2.13",
>   "org.scala-lang.modules" % "scala-collection-compat_2.13"
> )
> ```
> 
> Akka 2.6.17 is now published for Scala 3, but to compile akka-grpc against 3 `akka-http` needs to be published for Scala 3 as well

Yes, progress on Scala 3 artifacts for Akka HTTP is tracked in https://github.com/akka/akka-http/issues/3891 . Until that time the recommendation is to use `CrossVersion.for3Use2_13` for using Akka gRPC in a Scala 3 project.
